File: course_101/exercises/easy_6/combining_arrays.rb
# Write a method that takes two Arrays as arguments, and returns an Array that contains all of the values from the argument Arrays.
# There should be no duplication of values in the returned Array, even if there are duplicates in the original Arrays.

# solution 1
def merge(arr1, arr2)
  (arr1 + arr2).uniq
end

# solution 2
def merge(arr1, arr2)
  merged = []
  arr1.each { |object| merged << object}
  arr2.each { |object| merged << object if !merged.include?(object)}
  merged
end

# solution 3
def merge(arr1, arr2)
  arr1 | arr2 # set union - Ruby method for combining two arrays with no duplicates
end

puts merge([1, 3, 5], [3, 6, 9]) == [1, 3, 5, 6, 9]

File: course_js210_computational_thinking_and_javascript_programming/exercises/js210_small_problems/easy_4/combining_arrays.js
// Write a function that takes two arrays as arguments, and returns an array containing the union of the values from the two. There should be no duplication of values in the returned array, even if there are duplicates in the original arrays. You may assume that both arguments will always be arrays.

/*
- concat arrays
- select any value so it only appears once in new array
*/


function isUnique(element, index, array) {
  return array.indexOf(element) === index;
}

function concatUniqe(arr1, arr2) {
  return arr1.concat(arr2).filter(isUnique)
}

console.log(concatUniqe([1, 3, 5], [3, 6, 9]));    // [1, 3, 5, 6, 9]
